lib/preprocess.js

```diff
@@ -0,0 +1,42 @@
+import assert from 'node:assert'
+import { ethers } from 'ethers'
+
+export const preprocess = async ({
+  rounds,
+  cid,
+  roundIndex,
+  web3Storage,
+  logger
+}) => {
+  const measurements = (await fetchMeasurements(web3Storage, cid))
+    .filter(measurement => {
+      try {
+        assertValidMeasurement(measurement)
+        return true
+      } catch (err) {
+        logger.error('Invalid measurement', err.message, measurement)
```

Comment:
How can we gain visibility into this? Can we e.g. create a Grafana chart showing us the number (or fraction) of invalid measurements per committed batch or per SPARK round?

Reply:
It's probably best to improve observability in a follow-up PR though.

Reply:
https://github.com/filecoin-station/spark-evaluate/issues/2